discovery_engine_core/web-api2/src/elastic.rs

```diff
@@ -65,18 +76,97 @@ impl Default for Config {
 }
 
 pub(crate) struct ElasticSearchClient {
-    #[allow(dead_code)]
     config: Config,
-    #[allow(dead_code)]
+    url_to_index: Url,
     client: reqwest::Client,
 }
 
 impl ElasticSearchClient {
-    pub(crate) fn new(config: Config) -> Self {
-        Self {
+    pub(crate) fn new(config: Config) -> Result<Self, SetupError> {
+        let mut url_to_index: Url = config.url.parse()?;
+        url_to_index
+            .path_segments_mut()
+            .map_err(|()| "non segmentable url in config")?
+            .push(&config.index_name);
+
+        Ok(Self {
             config,
+            url_to_index,
             client: reqwest::Client::new(),
+        })
+    }
+
+    pub(crate) async fn delete_documents(&self, documents: &[DocumentId]) -> Result<(), Error> {
+        if documents.is_empty() {
+            return Ok(());
         }
+
+        let response = self
+            .bulk_request(
+                documents
+                    .iter()
+                    .map(|document_id| json!({ "delete": { "_id": document_id }})),
+            )
+            .await?;
+
+        if response.errors {
+            let mut errors = HashMap::new();
+            for (idx, mut response) in response.items.into_iter().enumerate() {
+                if let Some(response) = response.remove("delete") {
+                    if let Ok(status) = StatusCode::from_u16(response.status) {
+                        if status != StatusCode::NOT_FOUND && !status.is_success() {
+                            errors.insert(documents[idx].clone(), response.error);
+                        }
+                    } else {
+                        error!("Non http status code: {}", response.status);
+                    }
+                } else {
+                    error!("Bulk delete request contains non delete responses: {response:?}");
+                }
+            }
+
+            if !errors.is_empty() {
+                return Err(FailedToDeleteSomeDocuments { errors }.into());
+            }
+        }
```

Comment:
:gray_exclamation: I'm really not sure if having that error handling makes sense.

- Non existing documents don't cause `response.errors`  to be `true`.
- The only way we can have some but not all document deletions failing are (AFIK) supper edge case failure cases in which it often 1) isn't possible to fix it by directly retrying 2) it isn't too big of a problem to retry all once the problem was fixed.